plutus-benchmark/marlowe/src/PlutusBenchmark/Marlowe/Core/V1/Semantics.hs

```diff
@@ -117,633 +107,741 @@ import PlutusTx.AssocMap qualified as Map
+    ReduceShadowing ValueId Integer Integer
+  | --                 oldVal ^  newVal ^
+    ReduceAssertionFailed
   deriving stock (Haskell.Show, Data)
 
-
 -- | Result of 'reduceContractStep'
-data ReduceStepResult = Reduced ReduceWarning ReduceEffect State Contract
-                      | NotReduced
-                      | AmbiguousTimeIntervalReductionError
+data ReduceStepResult
+  = Reduced ReduceWarning ReduceEffect State Contract
+  | NotReduced
+  | AmbiguousTimeIntervalReductionError
   deriving stock (Haskell.Show, Data)
 
-
 -- | Result of 'reduceContractUntilQuiescent'
-data ReduceResult = ContractQuiescent Bool [ReduceWarning] [Payment] State Contract
-                  | RRAmbiguousTimeIntervalError
+data ReduceResult
+  = ContractQuiescent Bool [ReduceWarning] [Payment] State Contract
+  | RRAmbiguousTimeIntervalError
   deriving stock (Haskell.Show, Data)
 
-
 -- | Warning of 'applyCases'
-data ApplyWarning = ApplyNoWarning
-                  | ApplyNonPositiveDeposit Party AccountId Token Integer
+data ApplyWarning
+  = ApplyNoWarning
+  | ApplyNonPositiveDeposit Party AccountId Token Integer
   deriving stock (Haskell.Show, Data)
 
-
 -- | Result of 'applyCases'
-data ApplyResult = Applied ApplyWarning State Contract
-                 | ApplyNoMatchError
-                 | ApplyHashMismatch
+data ApplyResult
+  = Applied ApplyWarning State Contract
+  | ApplyNoMatchError
+  | ApplyHashMismatch
   deriving stock (Haskell.Show, Data)
 
-
 -- | Result of 'applyAllInputs'
-data ApplyAllResult = ApplyAllSuccess Bool [TransactionWarning] [Payment] State Contract
-                    | ApplyAllNoMatchError
-                    | ApplyAllAmbiguousTimeIntervalError
-                    | ApplyAllHashMismatch
+data ApplyAllResult
+  = ApplyAllSuccess Bool [TransactionWarning] [Payment] State Contract
+  | ApplyAllNoMatchError
+  | ApplyAllAmbiguousTimeIntervalError
+  | ApplyAllHashMismatch
   deriving stock (Haskell.Show, Data)
 
-
 -- | Warnings during transaction computation
-data TransactionWarning = TransactionNonPositiveDeposit Party AccountId Token Integer
-                        | TransactionNonPositivePay AccountId Payee Token Integer
-                        | TransactionPartialPay AccountId Payee Token Integer Integer
---                                                 ^ src    ^ dest     ^ paid   ^ expected
-                        | TransactionShadowing ValueId Integer Integer
---                                                 oldVal ^  newVal ^
-                        | TransactionAssertionFailed
+data TransactionWarning
+  = TransactionNonPositiveDeposit Party AccountId Token Integer
+  | TransactionNonPositivePay AccountId Payee Token Integer
+  | TransactionPartialPay AccountId Payee Token Integer Integer
+  | --                    ^ src    ^ dest     ^ paid   ^ expected
+    TransactionShadowing ValueId Integer Integer
+  | --                       oldVal ^  newVal ^
```

Comment:
Oof these comments are so weird, but that's nothing to do with your changes, I'm just complaining...